flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java

```diff
@@ -262,6 +251,40 @@ public void failJobDueToTaskFailure(Throwable cause, ExecutionAttemptID failingT
 		}
 	}
 
+	@Test
+	public void testExpiredCheckpointExceedsTolerableFailureNumber() {
+		// create some mock Execution vertices that receive the checkpoint trigger messages
+		ExecutionVertex vertex1 = mockExecutionVertex(new ExecutionAttemptID());
+		ExecutionVertex vertex2 = mockExecutionVertex(new ExecutionAttemptID());
+
+		final String errorMsg = "Exceeded checkpoint failure tolerance number!";
+		CheckpointFailureManager checkpointFailureManager = getCheckpointFailureManager(errorMsg);
+		CheckpointCoordinator coord = getCheckpointCoordinator(new JobID(), vertex1, vertex2, checkpointFailureManager);
+
+		try {
+			// trigger the checkpoint. this should succeed
+			final CompletableFuture<CompletedCheckpoint> checkPointFuture = coord.triggerCheckpoint(false);
+			manuallyTriggeredScheduledExecutor.triggerAll();
+			assertFalse(checkPointFuture.isCompletedExceptionally());
+
+			coord.abortPendingCheckpoints(new CheckpointException(CHECKPOINT_EXPIRED));
+
+			fail("Test failed.");
+		}
+		catch (Exception e) {
+			//expected
+			assertTrue(e instanceof RuntimeException);
+			assertEquals(errorMsg, e.getMessage());
```

Comment:
Hmm, I think different people have quite different tastes :-)

I am doing it so because the rest of the tests are using the `try-catch-fail` style. But I am fine with either way? 

Do you insist on this? I can make the change if you do :-)

Reply:
No, I don't insist :)

Reply:
Ha, then I will mark this as resolved.